### PR TITLE
[HTML5] Add PWA support to the editor page.

### PR DESCRIPTION
**Pull request by Faless**
This allows to install it as an app, and provide offline support (after
the first run).
Practically, this boils down to adding a JSON file as a manifest, an
offline page to be displayed when the cached files are not avaialble,
and a JS file to cache resources and return them.

The reason for the "first run requirements" is that some browsers, will
emit an "install" by just visiting the page (to see if the JS code is
compatibile), and we do not want to force casual visitors to just
download the 10 MiB+ compressed editor WebAssembly file without pressing
the start button.

Special thanks to Hugo Locurcio (Calinou) for the initial work.

https://github.com/Calinou/godot/commit/2aad127ab3c948a6458b7498c03125453b38b558

### CODE REVIEW

**Comment by Faless** [comment]
Note: I also moved the `Zip` template generation into the helper file, since the SCsub was getting really messy.

File: platform/javascript/js/libs/library_godot_audio.js [hunk 1]
```diff
@@ -250,7 +250,7 @@ const GodotAudioWorklet = {
 					GodotAudioWorklet.worklet = null;
 					GodotAudioWorklet.promise = null;
 					resolve();
-				}).catch(function(err) { /* aborted? */});
+				}).catch(function (err) { /* aborted? */ });
```
Comment (akien-mga) [inline platform/javascript/js/libs/library_godot_audio.js:253]: This should be amended in the first commit if we want to be thorough ;)

Comment (Faless) [reply platform/javascript/js/libs/library_godot_audio.js:253]: righty, hasty me, thanks for cheking

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**